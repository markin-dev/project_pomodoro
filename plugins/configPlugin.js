import * as config from '../constatns';

export default {
    install(Vue) {
        // eslint-disable-next-line
        Vue.prototype.$getConfig = (key) => config[key];
    },
};
