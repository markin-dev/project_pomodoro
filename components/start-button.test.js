// изменение констант не должно влиять на тесты

import { shallowMount, createLocalVue } from '@vue/test-utils';
import startButton from './start-button.vue';
import {
    TIMER_SIZE,
    BLUE,
    LIGHT_BLUE,
} from '../constatns';
import configPlugin from '../plugins/configPlugin';

describe('start-button.vue', () => {
    it('has the expected html structure', () => {
        const localVue = createLocalVue();

        localVue.use(configPlugin);

        const wrapper = shallowMount(startButton, {
            localVue,
            propsData: {
                radius: TIMER_SIZE,
                color: BLUE,
                hoverColor: LIGHT_BLUE,
            },
        });

        expect(wrapper.element).toMatchSnapshot();
    });
});
