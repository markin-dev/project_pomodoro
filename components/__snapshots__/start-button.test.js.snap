// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`start-button.vue has the expected html structure 1`] = `
<svg
  class="start-button"
>
  <circle
    cx="110"
    cy="110"
    fill="#42A5F5"
    r="220"
    style="fill: #42A5F5;"
  />
   
  <svg
    fill="none"
    height="88"
    viewBox="0 0 69 88"
    width="88"
    x="74.80000000000001"
    y="66"
  >
    <path
      d="M0.92453 0.4151V87.1698L69 43.7925L0.92453 0.4151Z"
      fill="#EEEEEE"
    />
  </svg>
</svg>
`;
